src/coverlet.core/Instrumentation/ModuleTrackerTemplate.cs

```diff
@@ -72,64 +74,93 @@ public static void RecordSingleHit(int hitLocationIndex)
-                                $"{HitsFilePath} has {hitsLength} entries but on memory {nameof(HitsArray)} has {hitsArray.Length}");
-                        }
+                            int hitsLength = br.ReadInt32();
+                            if (hitsLength != hitsArray.Length)
+                            {
+                                throw new InvalidOperationException(
+                                    $"{HitsFilePath} has {hitsLength} entries but on memory {nameof(HitsArray)} has {hitsArray.Length}");
+                            }
 
-                        for (int i = 0; i < hitsLength; ++i)
-                        {
-                            int oldHitCount = br.ReadInt32();
-                            bw.Seek(-sizeof(int), SeekOrigin.Current);
-                            if (SingleHit)
-                                bw.Write(hitsArray[i] + oldHitCount > 0 ? 1 : 0);
-                            else
-                                bw.Write(hitsArray[i] + oldHitCount);
+                            for (int i = 0; i < hitsLength; ++i)
+                            {
+                                int oldHitCount = br.ReadInt32();
+                                bw.Seek(-sizeof(int), SeekOrigin.Current);
+                                if (SingleHit)
+                                    bw.Write(hitsArray[i] + oldHitCount > 0 ? 1 : 0);
+                                else
+                                    bw.Write(hitsArray[i] + oldHitCount);
+                            }
                         }
                     }
+
+                    // On purpose this is not under a try-finally: it is better to have an exception if there was any error writing the hits file
+                    // this case is relevant when instrumenting corelib since multiple processes can be running against the same instrumented dll.
+                    mutex.ReleaseMutex();
+                    WriteLog($"Hit file '{HitsFilePath}' flushed, size {new FileInfo(HitsFilePath).Length}");
                 }
+            }
+            catch (Exception ex)
+            {
+                WriteLog(ex.ToString());
+                throw;
+            }
+        }
 
-                // On purpose this is not under a try-finally: it is better to have an exception if there was any error writing the hits file
-                // this case is relevant when instrumenting corelib since multiple processes can be running against the same instrumented dll.
-                mutex.ReleaseMutex();
+        private static void WriteLog(string logText)
+        {
+            if (_enableLog)
+            {
+                try
+                {
+                    // We don't set path as global var to keep benign possible errors inside try/catch
+                    // I'm not sure that location will be ok in every scenario
+                    string location = Assembly.GetExecutingAssembly().Location;
```

Comment:
This looks like it will write the log file into the `%userprofile%\.nuget\packages` location where the coverlet package was unpacked. IMO that's a terrible place to accumulate or discover log files.
Generally log files go into the `IntermediateOutputPath`  variable. Can you plumb through that msbuild property to this code?

Reply:
Actually this class is injected inside instrumented lib so the log file is generated on output(debug/release) folder.

Reply:
oh, ok. Sounds good.